fixtureeditor/main.cpp

```diff
@@ -52,13 +52,23 @@ void printVersion()
 {
     QTextStream cout(stdout, QIODevice::WriteOnly);
 
+#if QT_VERSION >= QT_VERSION_CHECK( 5, 15, 0 )
+    cout << Qt::endl;
+    cout << App::longName() << " " << App::version() << Qt::endl;
```

Comment:
I don't like this solution.
There has to be another way. Like "\n" at each line and a single flush at the end.

Reply:
are you kidding? I just replaced endl by Qt;;endl, what's so bad about that, the rest is your own code

Reply:
No, I'm not kidding. I don't like the code duplication just to change a constant.
If we go that way, every time we add an option, we need to remember to add it twice, which is ugly.
Right now I don't have the chance to look into this myself, but I will pursue a unified solution I proposed before.

Reply:
ok i got you wrong, sorry
I see what you mean now, and agree it's ugly. I simply followed the suggestion from the error message:
`error: ‘QTextStream& QTextStreamFunctions::endl(QTextStream&)’ is deprecated: Use Qt::endl [-Werror=deprecated-declarations]`

I just tried with `<< '\n' << flush;` but that leads to the same situation, it wants me to use Qt::flush instead. which like Qt::endl (I think, can't find earlier references) has been introduced with 5.15. 

Also I don't really understand why in the beginning of my porting effort I didn't get these errors at all, there were only a handful regarding the random stuff etc. from the first couple of commits. And just when I thought I was done, all these enum constant errors and finally this stuff popped up... and why the other guy in the forum didn't have to do all that... ?

Reply:
>  and why the other guy in the forum didn't have to do all that... ?

I think he builds on macOS with CLANG. Most likely it has different build flags enabled than GCC

Reply:
Here's an idea to avoid duplication, not sure if you will lke this:

`#if QT_VERSION >= QT_VERSION_CHECK( 5, 15, 0 )
#define endl Qt::endl
#endif

// ... printVersion() and printUsage() definitions

#if QT_VERSION >= QT_VERSION_CHECK( 5, 15, 0 )
#undef endl
#endif
`

Reply:
Maybe something along these lines:
```
#if QT_VERSION >= QT_VERSION_CHECK( 5, 15, 0 )
using line_end = Qt::endl;
#else
using line_end = endl;
#endif
...
cout << "..." << line_end;
```
Or maybe just:
```
#if QT_VERSION >= QT_VERSION_CHECK( 5, 15, 0 )
using endl = Qt::endl;
#endif
```

we're using c++, so preprocesor is not necessary for that.
I'm not sure where does `endl` without namespace come from, so I'm not sure if it will work. Is there somewhere `using namespace` that disappears in newer version of Qt?

BTW. I didn't look through all the code here, just this one bit got my interest when I was reading through the emails.

Reply:
endl (and flush) without namespace comes from QTextStreamFunctions according to the error I quoted above, and that has been deprecated in Qt 5.15...

Reply:
Alright, I checked this.
@edogawa23  Please proceed with
```
#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
#define endl Qt::endl
#endif
```
Apparently undef is not required.
Please also remove spaces from QT_VERSION_CHECK usage, like above, so it looks like the existing code.
Thanks